addons/persistence/infinispan-persistence-addon/src/main/java/org/kie/kogito/infinispan/ProtoStreamObjectMarshallingStrategy.java

```
@@ -91,6 +96,28 @@ public void registerMarshaller(MessageMarshaller<?>... marshallers) {
             typeToClassMapping.putIfAbsent(marshaller.getTypeName(), marshaller.getJavaClass());
         }
     }
+    
+    @Override
+	 public String marshalToJson(Object object) {
+		String json = null;
+		try {
+			json = MAPPER.writeValueAsString(object);
+		} catch (JsonProcessingException e) {
+			 LOGGER.error("Error while writing object as json", e);
```

I think it should throw an exception and not just silently fail with an error log.